homeassistant/components/radarr/__init__.py

```diff
@@ -70,7 +70,16 @@ async def async_setup_entry(hass: HomeAssistant, entry: RadarrConfigEntry) -> bo
         status=StatusDataUpdateCoordinator(hass, host_configuration, radarr),
     )
     for field in fields(data):
-        await getattr(data, field.name).async_config_entry_first_refresh()
+        coordinator: RadarrDataUpdateCoordinator = getattr(data, field.name)
+        # Movie update can take a while depending on Radarr database size
+        if field.name == "movie":
+            entry.async_create_background_task(
+                hass,
+                coordinator.async_config_entry_first_refresh(),
```

Comment:
Wait! This function raises exceptions that is not handled. That function is meant to run inline with setup to habdle not ready events. @frenck ?

You should use regular refresh for this. Just trigger a standard coordinator refresh.

Reply:
> Wait! This function raises exceptions that is not handled. 

Correct, but there are many more called here, still handling all other cases.

> You should use regular refresh for this. Just trigger a standard coordinator refresh.

I thought about that too for a bit, but... I think this works fine too right here.

Reply:
I don't follow "many more"? async_create_background_task will not catch any exceptions. All those leak to event loop with backtraces.